src/storage/src/table/cell_based_table.rs

```diff
@@ -34,16 +34,96 @@ use super::mem_table::RowOp;
 use super::TableIter;
 use crate::cell_based_row_deserializer::{CellBasedRowDeserializer, ColumnDescMapping};
 use crate::cell_based_row_serializer::CellBasedRowSerializer;
+use crate::cell_serializer::CellSerializer;
+use crate::dedup_pk_cell_based_row_serializer::DedupPkCellBasedRowSerializer;
 use crate::error::{StorageError, StorageResult};
 use crate::keyspace::StripPrefixIterator;
 use crate::storage_value::{StorageValue, ValueMeta};
 use crate::{Keyspace, StateStore, StateStoreIter};
 
+pub type DedupPkCellBasedTable<S> = CellBasedTableExtended<S, DedupPkCellBasedRowSerializer>;
+
+impl<S: StateStore> DedupPkCellBasedTable<S> {
+    pub fn new_dedup_pk_cell_based_table(
+        keyspace: Keyspace<S>,
+        columns: Vec<ColumnDesc>,
+        order_types: Vec<OrderType>,
+        pk_indices: Vec<usize>,
+        dist_key_indices: Option<Vec<usize>>,
+    ) -> Self {
+        let column_ids = columns.iter().map(|c| c.column_id).collect_vec();
+        let cell_based_row_serializer =
+            DedupPkCellBasedRowSerializer::new(&pk_indices, &columns, &column_ids);
+
+        Self::new_extended(
```

Comment:
add docs, mention how everything should instantiate the same way, except for cell based row serializer.